diff --git a/src/engine.h b/src/engine.h
index f3c78398..70166347 100644
--- a/src/engine.h
+++ b/src/engine.h
@@ -90,6 +90,18 @@ class Engine {
     void load_small_network(const std::string& file);
     void save_network(const std::pair<std::optional<std::string>, std::string> files[2]);
 
+
+    void load_big_network(std::istream& is) {
+        networks.modify_and_replicate([this, &is](auto& networks_) { networks_.big.load(is); });
+        threads.clear();
+        threads.ensure_network_replicated();
+    }
+
+    void load_small_network(std::istream& is) {
+        networks.modify_and_replicate([this, &is](auto& networks_) { networks_.small.load(is); });
+        threads.clear();
+        threads.ensure_network_replicated();
+    }
     // utility functions
 
     void trace_eval() const;
diff --git a/src/main.cpp b/src/main.cpp
index a6a3d1c4..3af2a48c 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -21,12 +21,14 @@
 #include "bitboard.h"
 #include "misc.h"
 #include "position.h"
+#include "tune.h"
 #include "types.h"
 #include "uci.h"
-#include "tune.h"
 
 using namespace Stockfish;
 
+UCIEngine uci_global(0, nullptr);
+
 int main(int argc, char* argv[]) {
 
     std::cout << engine_info() << std::endl;
@@ -34,11 +36,9 @@ int main(int argc, char* argv[]) {
     Bitboards::init();
     Position::init();
 
-    UCIEngine uci(argc, argv);
-
-    Tune::init(uci.engine_options());
+    Tune::init(uci_global.engine_options());
 
-    uci.loop();
+    uci_global.loop();
 
     return 0;
 }
diff --git a/src/nnue/network.h b/src/nnue/network.h
index 15208255..7f9f7dfa 100644
--- a/src/nnue/network.h
+++ b/src/nnue/network.h
@@ -78,9 +78,12 @@ class Network {
 
     void initialize();
 
-    bool                       save(std::ostream&, const std::string&, const std::string&) const;
+    bool save(std::ostream&, const std::string&, const std::string&) const;
+
+   public:
     std::optional<std::string> load(std::istream&);
 
+   private:
     bool read_header(std::istream&, std::uint32_t*, std::string*) const;
     bool write_header(std::ostream&, std::uint32_t, const std::string&) const;
 
diff --git a/src/uci.cpp b/src/uci.cpp
index 9b60680d..76afd27c 100644
--- a/src/uci.cpp
+++ b/src/uci.cpp
@@ -37,6 +37,8 @@
 #include "types.h"
 #include "ucioption.h"
 
+std::string js_getline();  // src/dualnet-glue.cpp
+
 namespace Stockfish {
 
 constexpr auto StartFEN = "rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1";
@@ -62,7 +64,7 @@ void UCIEngine::print_info_string(const std::string& str) {
 
 UCIEngine::UCIEngine(int argc, char** argv) :
     engine(argv[0]),
-    cli(argc, argv) {
+    cli(0, nullptr) {
 
     engine.get_options().add_info_listener([](const std::optional<std::string>& str) {
         if (str.has_value())
@@ -88,6 +90,8 @@ void UCIEngine::loop() {
             && !getline(std::cin, cmd))  // Wait for an input or an end-of-file (EOF) indication
             cmd = "quit";
 
+        cmd = js_getline();
+
         std::istringstream is(cmd);
 
         token.clear();  // Avoid a stale if getline() returns nothing or a blank line
@@ -164,7 +168,7 @@ void UCIEngine::loop() {
             sync_cout << "Unknown command: '" << cmd << "'. Type help for more information."
                       << sync_endl;
 
-    } while (token != "quit" && cli.argc == 1);  // The command-line arguments are one-shot
+    } while (token != "quit");  // The command-line arguments are one-shot
 }
 
 Search::LimitsType UCIEngine::parse_limits(std::istream& is) {
diff --git a/src/uci.h b/src/uci.h
index 23745f96..88cf29dc 100644
--- a/src/uci.h
+++ b/src/uci.h
@@ -54,8 +54,9 @@ class UCIEngine {
 
     auto& engine_options() { return engine.get_options(); }
 
+    Engine engine;
+
    private:
-    Engine      engine;
     CommandLine cli;
 
     static void print_info_string(const std::string& str);
